widget/embedded/src/components/Filter/Filter.tsx

```diff
@@ -0,0 +1,41 @@
+import type { PropTypes } from './Filter.type';
+
+import { FilterIcon, Popover } from '@rango-dev/ui';
+import React, { useState } from 'react';
+
+import { getContainer } from '../../utils/common';
+
+import { Badge, Button, IconContainer } from './Filter.style';
+import { FilterContent } from './FilterContent';
+
+export function Filter(props: PropTypes) {
```

Comment:
Don't you think that it is better to make this a general stateless component which accepts items, selectedItem, open and other things as props? Also the name `filter` can be misleading because it does not make it clear that this is a button or a list or a container.